pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -137,6 +151,16 @@ func (rc *reconciler) reconcile() {
 	for _, attachedVolume := range rc.actualStateOfWorld.GetAttachedVolumes() {
 		if !rc.desiredStateOfWorld.VolumeExists(
 			attachedVolume.VolumeName, attachedVolume.NodeName) {
+
+			// Don't even try to start an operation if there is already one running
+			// This check must be done before we do any other checks, as otherwise the other checks
+			// may pass while at the same time the volume leaves the pending state, resulting in
+			// double detach attempts
+			if rc.attacherDetacher.IsOperationPending(attachedVolume.VolumeName, "") {
```

Comment:
I am not sure the logic here. If all the checks pass, it will trigger DetachVolume operation through operation_executor in which it has the logic to prevent two operations working on the same volume (in nestedPendingOperation.go). Why we need an extra check for isOperationPending here?